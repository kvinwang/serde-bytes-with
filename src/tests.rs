#[test]
fn test_expand() {
    let stream = crate::patch(syn::parse_quote! {
        #[derive(Serialize, Deserialize, TypeInfo)]
        #[derive(Clone, PartialEq, Message)]
        pub struct Attestation {
            #[prost(int32, tag = "1")]
            pub version: i32,
            #[prost(string, tag = "2")]
            pub provider: String,
            #[prost(message, optional, tag = "3")]
            pub payload: Option<AttestationReport>,
            #[prost(bytes = "vec", tag = "5")]
            pub encoded_report: Vec<u8>,
            #[prost(bytes = "vec", optional, tag = "6")]
            pub timestamp: Option<Vec<u8>>,
            #[prost(bytes = "vec", repeated, tag = "7")]
            pub timestamp: Vec<Vec<u8>>,
        }
    });
    insta::assert_snapshot!(rustfmt_snippet::rustfmt_token_stream(&stream).unwrap())
}
