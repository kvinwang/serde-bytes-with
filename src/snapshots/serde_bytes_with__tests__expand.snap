---
source: src/tests.rs
assertion_line: 22
expression: "rustfmt_snippet::rustfmt_token_stream(&stream).unwrap()"
---
#[derive(Serialize, Deserialize, TypeInfo, Clone, PartialEq, Message)]
pub struct Attestation {
    #[prost(int32, tag = "1")]
    pub version: i32,
    #[prost(string, tag = "2")]
    pub provider: ::prost::alloc::string::String,
    #[prost(message, optional, tag = "3")]
    pub payload: ::core::option::Option<AttestationReport>,
    #[prost(bytes = "vec", tag = "5")]
    #[serde(with = "as_hexed_bytes")]
    pub encoded_report: ::prost::alloc::vec::Vec<u8>,
    #[prost(bytes = "vec", tag = "6")]
    #[serde(with = "as_hexed_bytes")]
    pub timestamp: ::prost::alloc::vec::Vec<u8>,
}
